dotnet/src/SemanticKernel.Abstractions/AI/XmlPromptParser.cs

```diff
@@ -37,7 +39,11 @@ public static bool TryParse(string prompt, [NotNullWhen(true)] out List<PromptNo
             return false;
         }
 
-        var xmlDocument = new XmlDocument();
+        var xmlDocument = new XmlDocument()
+        {
+            PreserveWhitespace = true
```

Comment:
Can you add a comment to explain why we need to preserve whitespace?

Reply:
Whitespace may be significant if for example we want the LLM to respond using a similar formatting. This change was requested by a team that sends well formatted code to the LLM and wants similarly formatted code returned.

Reply:
Consider adding these examples you mention in the chat to the comment.  I think they are valuable in understanding why the whitespace is significant.